games-roguelike/nethack/nethack-3.6.1.ebuild

```diff
@@ -0,0 +1,129 @@
+HOMEPAGE="https://www.nethack.org/"
+SRC_URI="https://nethack.org/download/${PV}/nethack-${MY_PV}-src.tgz"
+
+LICENSE="nethack"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~hppa ~x86 ~x86-fbsd"
+IUSE="X"
+
+RDEPEND="sys-libs/ncurses:0=
+	X? (
+		x11-libs/libXaw
+		x11-libs/libXpm
+		x11-libs/libXt
+	)"
+DEPEND="${RDEPEND}
+	X? ( x11-base/xorg-proto )
+	"
+BDEPEND="virtual/pkgconfig
+	X? (
+		x11-apps/bdftopcf
+		x11-apps/mkfontdir
+	)"
+
+NETHACK_GROUP="gamestat"
+
+pkg_setup() {
+	HACKDIR="/usr/$(get_libdir)/${PN}"
+
+	enewgroup gamestat 36
+}
+
+src_prepare() {
+	eapply "${FILESDIR}/${P}-recover.patch"
+	eapply_user
+
+	cp "${FILESDIR}/${PN}-3.6.0-hint-$(usex X x11 tty)" hint || die "Failed to copy hint file"
+	sys/unix/setup.sh hint || die "Failed to run setup.sh"
+}
+
+src_compile() {
+	append-cflags -I../include -DDLB -DSECURE -DLINUX -DTIMED_DELAY -DVISION_TABLES
+	append-cflags -DDUMPLOG -DSCORE_ON_BOTL -DSTATUS_HILITES
+	append-cflags '-DCOMPRESS=\"${EPREFIX}/bin/gzip\"' '-DCOMPRESS_EXTENSION=\".gz\"'
+	append-cflags "-DHACKDIR=\\\"${EPREFIX}${HACKDIR}\\\"" "-DVAR_PLAYGROUND=\\\"${EPREFIX}/var/games/nethack\\\""
+	append-cflags "-DDEF_PAGER=\\\"${PAGER}\\\""
+	append-cflags -DSYSCF "-DSYSCF_FILE=\\\"${EPREFIX}/etc/nethack.sysconf\\\""
+
+	use X && append-cflags -DX11_GRAPHICS -DUSE_XPM
+
+	LOCAL_MAKEOPTS=(
+		CC="$(tc-getCC)" CFLAGS="${CFLAGS}" LFLAGS="${LDFLAGS}"
+		WINTTYLIB="$($(tc-getPKG_CONFIG) --libs ncurses)"
+		HACKDIR="${EPREFIX}${HACKDIR}" INSTDIR="${ED}/${HACKDIR}"
+		SHELLDIR="${ED}/usr/bin" VARDIR="${ED}/var/games/nethack"
+		)
+
+	emake "${LOCAL_MAKEOPTS[@]}" nethack recover Guidebook spec_levs
+
+	# Upstream still has some parallel compilation bugs
+	emake -j1 "${LOCAL_MAKEOPTS[@]}" all
+}
+
+src_install() {
+	emake "${LOCAL_MAKEOPTS[@]}" install
+
+	mv "${ED}/${HACKDIR}/recover" "${ED}/usr/bin/recover-nethack" || die "Failed to move recover-nethack"
+
+	doman doc/nethack.6
+	newman doc/recover.6 recover-nethack.6
+	dodoc doc/Guidebook.txt
+
+	insinto /etc
+	newins sys/unix/sysconf nethack.sysconf
+
+	insinto /etc/skel
+	newins "${FILESDIR}/${PN}-3.6.0-nethackrc" .nethackrc
+
+	if use X ; then
+		cd "${S}/win/X11" || die "Failed to enter win/X11 directory"
+
+		# copy nethack x application defaults
+		insinto /etc/X11/app-defaults
+		newins NetHack.ad NetHack
+		rm "${ED}/${HACKDIR}/NetHack.ad" || die "Failed to remove NetHack.ad"
+
+		newicon nh_icon.xpm nethack.xpm
+		make_desktop_entry ${PN} Nethack
+
+		# install nethack fonts
+		bdftopcf -o nh10.pcf nh10.bdf || die "Converting fonts failed"
+		bdftopcf -o ibm.pcf ibm.bdf || die "Converting fonts failed"
+		insinto "${HACKDIR}/fonts"
+		doins *.pcf
+		cd "${ED}/${HACKDIR}/fonts" || die "Failed to enter fonts directory"
+		mkfontdir || die "The action mkfontdir ${HACKDIR}/fonts failed"
+	fi
+
+	rm -r "${ED}/var/games/nethack" || die "Failed to clean var/games/nethack"
+	keepdir /var/games/nethack/save
+
+	fowners -R "root:${NETHACK_GROUP}" /var/games/nethack
```

Comment:
Ok, one more trick: if you move this and below to `pkg_preinst` (i.e. make it change the owner just before installing), you can also move `enewgroup` there and you will no longer require people to create the group while building binary packages.